File: apps/frontend/cors-proxy.js
const http = require('http');
const https = require('https');
const url = require('url');

const PORT = 8080;
const API_BASE = 'https://api.extended.exchange';

const server = http.createServer((req, res) => {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization, X-API-Key');
  
  // Handle preflight requests
  if (req.method === 'OPTIONS') {
    res.writeHead(200);
    res.end();
    return;
  }

  const reqUrl = url.parse(req.url);
  console.log(`${new Date().toISOString()} ${req.method} ${req.url}`);
  
  // Check if this is an API request
  if (reqUrl.pathname.startsWith('/api/v1/')) {
    // Proxy API requests
    const targetUrl = API_BASE + req.url;
    console.log(`Proxying to: ${targetUrl}`);
    
    const options = {
      method: req.method,
      headers: {
        ...req.headers,
        host: 'api.extended.exchange'
      }
    };
    
    const proxyReq = https.request(targetUrl, options, (proxyRes) => {
      // Set CORS headers on the proxied response
      res.setHeader('Access-Control-Allow-Origin', '*');
      res.writeHead(proxyRes.statusCode, proxyRes.headers);
      proxyRes.pipe(res);
    });
    
    proxyReq.on('error', (e) => {
      console.error('Proxy error:', e.message);
      res.writeHead(502);
      res.end('Bad Gateway');
    });
    
    req.pipe(proxyReq);
    
  } else {
    // Serve static files from build/web
    const fs = require('fs');
    const path = require('path');
    
    let filePath = path.join(__dirname, 'build/web', reqUrl.pathname);
    
    // Default to index.html for SPA routing
    if (reqUrl.pathname === '/' || !path.extname(filePath)) {
      filePath = path.join(__dirname, 'build/web/index.html');
    }
    
    fs.readFile(filePath, (err, data) => {
      if (err) {
        res.writeHead(404);
        res.end('File not found');
        return;
      }
      
      // Set correct content type
      const ext = path.extname(filePath);
      const contentTypes = {
        '.html': 'text/html',
        '.js': 'application/javascript',
        '.css': 'text/css',
        '.json': 'application/json',
        '.png': 'image/png',
        '.jpg': 'image/jpeg',
        '.ico': 'image/x-icon',
        '.ttf': 'font/ttf',
        '.otf': 'font/otf'
      };
      
      const contentType = contentTypes[ext] || 'application/octet-stream';
      res.setHeader('Content-Type', contentType);
      res.writeHead(200);
      res.end(data);
    });
  }
});

server.listen(PORT, () => {
  console.log(`🚀 CORS Proxy Server running on http://localhost:${PORT}`);
  console.log(`📡 Proxying /api/v1/* requests to ${API_BASE}`);
  console.log(`📁 Serving static files from build/web/`);
});

File: apps/frontend/cors_proxy.py
#!/usr/bin/env python3
import http.server
import socketserver
import urllib.request
import urllib.parse
import os
import json
from urllib.error import HTTPError

PORT = 8080
API_BASE = 'https://api.extended.exchange'

class CORSProxyHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory='build/web', **kwargs)
    
    def end_headers(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization, X-API-Key')
        super().end_headers()
    
    def do_OPTIONS(self):
        self.send_response(200)
        self.end_headers()
        
    def do_GET(self):
        if self.path.startswith('/api/v1/'):
            self.proxy_request()
        else:
            super().do_GET()
    
    def do_POST(self):
        if self.path.startswith('/api/v1/'):
            self.proxy_request()
        else:
            self.send_error(404)
    
    def proxy_request(self):
        try:
            url = API_BASE + self.path
            print(f"Proxying {self.command} {self.path} -> {url}")
            
            # Prepare request
            req_headers = {}
            for header, value in self.headers.items():
                if header.lower() not in ['host', 'connection']:
                    req_headers[header] = value
            
            # Get request data for POST/PUT
            content_length = self.headers.get('Content-Length')
            data = None
            if content_length:
                data = self.rfile.read(int(content_length))
            
            # Make request
            request = urllib.request.Request(url, data=data, headers=req_headers, method=self.command)
            
            with urllib.request.urlopen(request) as response:
                self.send_response(response.status)
                
                # Copy response headers
                for header, value in response.headers.items():
                    if header.lower() not in ['connection', 'transfer-encoding']:
                        self.send_header(header, value)
                self.end_headers()
                
                # Copy response body
                self.wfile.write(response.read())
                
        except HTTPError as e:
            print(f"HTTP Error: {e.code} {e.reason}")
            self.send_error(e.code, e.reason)
        except Exception as e:
            print(f"Proxy Error: {e}")
            self.send_error(502, 'Bad Gateway')

if __name__ == "__main__":
    with socketserver.TCPServer(("", PORT), CORSProxyHandler) as httpd:
        print(f"🚀 CORS Proxy Server running on http://localhost:{PORT}")
        print(f"📡 Proxying /api/v1/* requests to {API_BASE}")
        print(f"📁 Serving static files from build/web/")
        httpd.serve_forever()